.github/workflows/test-playwright.yml

```diff
@@ -0,0 +1,158 @@
+        required: false
+      GITHUB_USER_NAME:
+        description: Username for the GitHub user configuration.
+        required: false
+      GITHUB_USER_SSH_KEY:
+        description: Private SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: false
+      NPM_REGISTRY_TOKEN:
+        description: Authentication for the private npm registry.
+        required: false
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+
+jobs:
+  run-playwright-test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    env:
+      PHP_CHECK: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up SSH
+        env:
+          GITHUB_USER_SSH_KEY: ${{ secrets.GITHUB_USER_SSH_KEY }}
+        if: ${{ env.GITHUB_USER_SSH_KEY != '' }}
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ env.GITHUB_USER_SSH_KEY }}
+
+      - name: Set up Git
+        env:
+          GITHUB_USER_EMAIL: ${{ secrets.GITHUB_USER_EMAIL }}
+          GITHUB_USER_NAME: ${{ secrets.GITHUB_USER_NAME }}
+        if: ${{ env.GITHUB_USER_EMAIL != '' && env.GITHUB_USER_NAME != '' }}
+        run: |
+          git config --global user.email "${{ env.GITHUB_USER_EMAIL }}"
+          git config --global user.name "${{ env.GITHUB_USER_NAME }}"
+
+      - name: Check if PHP installation is needed
+        run: |
+          if [ "${{ hashFiles('composer.json') }}" != "" ] && [ "${{ inputs.PHP_VERSION }}" != "" ]; then
+            echo "PHP_CHECK=true" >> $GITHUB_ENV
+          fi
+
+      - name: Set up PHP
+        if: ${{ env.PHP_CHECK == 'true' }}
+        uses: shivammathur/setup-php@v2
+        env:
+          COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: composer
+          coverage: none
+
+      - name: Install Composer dependencies
+        if: ${{ env.PHP_CHECK == 'true' }}
+        uses: ramsey/composer-install@v3
+        with:
+          composer-options: '--prefer-dist'
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Install Playwright dependencies
+        run: |
+          npx playwright install ${{ inputs.PLAYWRIGHT_BROWSER_ARGS }}
+
+      - name: Run script for test
+        continue-on-error: true
+        run: |
+          touch .env.ci
+          echo "${{ secrets.ENV_FILE_DATA }}" >> .env.ci
+          npm run ${{ inputs.SCRIPT_NAME }}
+
+      - name: Remove .env.ci file
+        run: |
+          rm .env.ci
```

Comment:
This is a very minor suggestion: While having this as an extra step surely works, especially since the previous step uses `continue-on-error: true`, an alternative approach could be to use a [`trap`](https://man7.org/linux/man-pages/man1/trap.1p.html) command in the previous step before `npm run`, similar to how we did it here: https://github.com/inpsyde/actions/blob/5d30147f89c39e066d3f9bea13ac08f1c5a96da9/setup-wireguard/action.yml#L16-L17.